### PR TITLE
Basic test env for selenium

### PR DESCRIPTION
**Pull request by stefanoborini**
Basic testing setup for selenium

### CODE REVIEW

**Comment by stefanoborini** [comment]
@kitchoi ready for review.

File: Makefile [hunk 1]
```diff
@@ -11,26 +11,49 @@ help:
...
+	sudo pip install --upgrade pip
 	sudo npm install -g configurable-http-proxy
-	python3 -mvenv venv
-	. venv/bin/activate && pip3 install docker-py
+
+.PHONY: pythondeps
+pythondeps:
+	@echo "Installing dependencies"
+	@echo "-----------------------"
+	if test -z "${NO_VENV}"; \
```
Comment (kitchoi) [inline Makefile]: If you just mean to see if the make file is being used on travis, you can check the `TRAVIS` environment variable. https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables

File: Makefile [hunk 2]
```diff
@@ -42,18 +65,37 @@ certs:
...
+	if test -z ${NO_VENV}; \
+	then \
+		. venv/bin/activate; \
+	fi; \
+	pushd jupyterhub; \
+        remoteappdb --db=remoteappmanager.db init;\
+        remoteappdb --db=remoteappmanager.db user create test; \
+        remoteappdb --db=remoteappmanager.db app create simphonyproject/simphonic-mayavi; \
+        remoteappdb --db=remoteappmanager.db app grant simphonyproject/simphonic-mayavi test; \
+        popd
```
Comment (kitchoi) [inline Makefile]: These 3 lines make the Makefile unsuitable for general use:
```
remoteappdb --db=remoteappmanager.db user create test;
remoteappdb --db=remoteappmanager.db app create simphonyproject/simphonic-mayavi;
remoteappdb --db=remoteappmanager.db app grant simphonyproject/simphonic-mayavi test;
```

File: Makefile (hunk 1 above)
Comment (stefanoborini) [inline Makefile]: No, I want to see if I need a venv or not. The fact that it's travis is accidental. Could be our jenkins.

**Comment by kitchoi** [comment]
The Makefile can be more useful for general use if the following is removed.  User of the script would be advised (in README/doc) to switch to a virtual environment before invoking certain targets.
```
+	if test -z ${NO_VENV}; \
 +	then \
 +		. venv/bin/activate; \
 +	fi; \\
```

File: dev-requirements.txt [hunk 3]
```diff
@@ -1,3 +1,3 @@
 flake8
-sphinx
```
Comment (kitchoi) [inline dev-requirements.txt]: why?

Comment (stefanoborini) [inline dev-requirements.txt]: it's a doc requirement, not a dev requirement.

Comment (kitchoi) [inline dev-requirements.txt]: I see.  Thanks.

**Comment by kitchoi** [comment]
Just a general comment/reminder/FYI that the novnc page's title can be modified as easily as passing a query parameter: `http://container_url/?titlle=whatever`

File: .travis.yml [hunk 4]
```diff
@@ -5,21 +5,33 @@ python:
...
-    - sphinx-build -W doc/source doc/build/sphinx
+    - export NO_VENV=1
+    - make test
+    - make docs
+    - make testdb
+    - make certs
+    - pushd jupyterhub
+    - mkdir -p /tmp/remoteapp
+    - chmod 755 `pwd`/start.sh
+    - "`pwd`/start.sh &"
```
Comment (kitchoi) [inline .travis.yml]: Just note that #146 would not be compatible here.

File: Makefile [hunk 5]
```diff
@@ -11,26 +11,49 @@ help:
...
-	@echo "Installing dependencies"
-	@echo "-----------------------"
+.PHONY: aptdeps
+aptdeps:
+	@echo "Installing apt dependencies"
+	@echo "---------------------------"
 	sudo apt-get update
-	sudo apt-get install docker-engine npm nodejs-legacy python3-pip python3.4-venv
+	sudo apt-get install -o Dpkg::Options::="--force-confold" --force-yes -y docker-engine npm nodejs-legacy python3-pip python3.4-venv
+	sudo pip install --upgrade pip
```
Comment (kitchoi) [inline Makefile]: Are you going to remove `sudo` here?

File: .travis.yml [hunk 6]
```diff
@@ -5,21 +5,32 @@ python:
...
 
 before_install:
-    - sudo apt-get update
-    - sudo apt-get install -o Dpkg::Options::="--force-confold" --force-yes -y docker-engine
-    - sudo pip install --upgrade pip
-    - pip install -r requirements.txt
-    - pip install -r dev-requirements.txt
-    - pip install -r doc-requirements.txt
+    - firefox --version
+    - make aptdeps
```
Comment (kitchoi) [inline .travis.yml]: sudo here?

**Comment by kitchoi** [comment]
Good to merge when it is green.  Thanks @stefanoborini

**Pull request merged**